### PR TITLE
frontend/feat/#1303/Improve auto-complete UI experience when selectin…

### PR DESCRIPTION
**Pull request by abhigyanshukla97**
…g Pickup/Drop by User

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Improve auto-complete UI experience for android.
video link -> https://drive.google.com/file/d/1275IKFkL3LFuvt1YsX0vbLThYwGuwDKQ/view?usp=sharing

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 1]
```diff
@@ -187,36 +185,36 @@ locationUnserviceableView state push =
 sourceDestinationImageView :: forall w. PrestoDOM (Effect Unit) w
 sourceDestinationImageView =
   frameLayout
-    [ height $ V 136
-    , width $ V 50
-    , margin (Margin 5 15 0 0)
-    , gravity CENTER
+    [ height $ V 100
+    , width $ V 35
+    , margin (Margin 0 9 0 0)
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:190]: @abhigyanshukla97 Use MarginTop

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 2]
```diff
@@ -241,19 +239,21 @@ sourceDestinationEditTextView state push =
...
-            [ height $ V 45
+            [ height $ V 37
             , weight 1.0
             , text state.source
-            , color Color.black800
-            , stroke if state.isSource == Just true && state.isSearchLocation == LocateOnMap then "1,#FDD836" else "0,#FDD836"
-            , background Color.white900
+            , color if not(state.isSource == Just true && state.isSearchLocation /= LocateOnMap) then "#A7A7A7" else Color.white900
+            , background "#303440"
+            , stroke "0,#000000"
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:251]: @abhigyanshukla97 If stroke is '0' you can remove

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 3]
```diff
@@ -241,19 +239,21 @@ sourceDestinationEditTextView state push =
...
             , fontStyle $ FontStyle.semiBold LanguageStyle
             , singleLine true
             , ellipsize true
-            , cornerRadius 10.0
-            , padding (Padding 5 0 5 0)
-            , margin (Margin 0 10 0 0)
+            , padding (Padding 8 7 32 7)
+            , margin (Margin 0 0 0 0)
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:256]: @abhigyanshukla97 Why Margin is required is everything is 0

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 4]
```diff
@@ -271,53 +271,48 @@ sourceDestinationEditTextView state push =
             , autoCorrectionType 1
             ]
         , linearLayout
-            [ height $ V 45
-            , width WRAP_CONTENT
+            [ height $ V 32
+            , width $ V 30
             , gravity CENTER
-            , padding (Padding 0 10 0 5)
+            , padding (Padding 0 10 0 2)
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:277]: Use PaddingVertical

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 5]
```diff
@@ -271,53 +271,48 @@ sourceDestinationEditTextView state push =
...
-              , cornerRadius 10.0
-              , stroke if state.isSource == Just false && state.isSearchLocation == LocateOnMap then "1,#FDD836" else "0,#FDD836"
-              , padding (Padding 5 0 5 0)
-              , background Color.white900
+              , color if (state.isSource == Just true && state.isSearchLocation /= LocateOnMap) then "#A7A7A7" else Color.white900
+              , stroke "0,#000000"
+              , padding (Padding 8 7 4 7)
+              , fontStyle $ FontStyle.semiBold LanguageStyle
               , hint (getString WHERE_TO)
               , hintColor "#A7A7A7"
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:317]: Can you add this color code in color.pur file and use

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 6]
```diff
@@ -271,53 +271,48 @@ sourceDestinationEditTextView state push =
...
         ]
         []
     , linearLayout
         [ height WRAP_CONTENT
         , width MATCH_PARENT
+        , cornerRadius 4.0
         , orientation HORIZONTAL
-        , margin (Margin 0 10 0 0)
+        , margin (Margin 0 12 0 0)
+        , background "#303440"
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:305]: Don't use direct color code

**Review by kavya-shree-s — COMMENTED** [review]
@abhigyanshukla97  Added comments ,resolve everywhere in the PR

**Review by kavya-shree-s — CHANGES_REQUESTED** [review]
@abhigyanshukla97 Added comments, resolve everywhere in the PR

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs (hunk 2 above)
Comment (abhigyanshukla97) [reply Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:251]: In EditText there is a blue underline to remove it we have to use this else it wont work.

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 7]
```diff
@@ -32,111 +32,110 @@ import Engineering.Helpers.Commons (getNewIDWithTag, os, safeMarginBottom, safeM
...
         (if os == "IOS" then [] else [ translateYAnimFromTop $ translateFullYAnimWithDurationConfig 500 ])
         $ linearLayout
          -- Temporary fix for iOS.
-            [ height MATCH_PARENT
+            [ height WRAP_CONTENT
             , width MATCH_PARENT
             , orientation VERTICAL
-            , padding (Padding 0 safeMarginTop 0 safeMarginBottom)
+            , background Color.black900
+            , padding $ Padding 0 safeMarginTop 0 16
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:67]: use paddingVertical

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 8]
```diff
@@ -32,111 +32,110 @@ import Engineering.Helpers.Commons (getNewIDWithTag, os, safeMarginBottom, safeM
...
                   , disableClickFeedback true
-                  , margin (Margin 16 10 16 0)
+                  , margin (Margin 20 21 0 0)
                   , gravity CENTER
                   ]
                   [ imageView
-                      [ height $ V 25
-                      , width $ V 25
+                      [ height $ V 30
+                      , width $ V 35
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:84]: reduce size and keep it more aligned to left, need more space for edit text

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 9]
```diff
@@ -241,19 +240,20 @@ sourceDestinationEditTextView state push =
...
+      , stroke if state.isSource == Just true && state.isSearchLocation == LocateOnMap then "1,#FDD836" else "0,#FDD836"
       ][ editText
-            [ height $ V 45
+            [ height $ V 37
             , weight 1.0
             , text state.source
-            , color Color.black800
-            , stroke if state.isSource == Just true && state.isSearchLocation == LocateOnMap then "1,#FDD836" else "0,#FDD836"
-            , background Color.white900
+            , color if not(state.isSource == Just true && state.isSearchLocation /= LocateOnMap) then "#A7A7A7" else Color.white900
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:250]: define color in `Color.purs` and then use

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 10]
```diff
@@ -271,54 +271,48 @@ sourceDestinationEditTextView state push =
...
+         [ editText
+            ( [ height $ V 37
               , weight 1.0
               , text state.destination
-              , color Color.black800
-              , cornerRadius 10.0
-              , stroke if state.isSource == Just false && state.isSearchLocation == LocateOnMap then "1,#FDD836" else "0,#FDD836"
-              , padding (Padding 5 0 5 0)
-              , background Color.white900
+              , color if (state.isSource == Just true && state.isSearchLocation /= LocateOnMap) then "#A7A7A7" else Color.white900
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:312]: define color in `Color.purs`

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs (hunk 9 above)
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:250]: here it should be `state.isSource == Just false` to highlight text

File: Frontend/ui-customer/src/Components/SearchLocationModel/View.purs [hunk 11]
```diff
@@ -271,54 +271,48 @@ sourceDestinationEditTextView state push =
...
+              , stroke $ "0," <> Color.black
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/SearchLocationModel/View.purs:313]: remove this

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

**Comment by ashishsingh101** [comment]
@ashkritidewan2000 check for current location text at Bottom in case of source it should be center aligned

**Comment by ashishsingh101** [comment]
check for yellow stroke in source in case of locate on map its not proper

**Comment by MercyQueen** [comment]
@ashkriti is this tested in IOS

**Comment by Vignesh-772** [comment]
closing as it is in another PR

**Pull request closed without merging**